ckan/views/feeds.py

```diff
@@ -0,0 +1,484 @@
+# encoding: utf-8
+
+import logging
+import urlparse
+
+from flask import Blueprint, make_response
+from werkzeug.exceptions import BadRequest
+from werkzeug.contrib.atom import AtomFeed
+
+from ckan.common import _, config, g, request, response, json
+import ckan.lib.helpers as h
+import ckan.lib.base as base
+import ckan.model as model
+import ckan.logic as logic
+import ckan.plugins as plugins
+
+
+log = logging.getLogger(__name__)
+
+feeds = Blueprint(u'feeds', __name__, url_prefix=u'/feeds')
+
+ITEMS_LIMIT = config.get(u'ckan.feeds.limit', 20)
+BASE_URL = config.get(u'ckan.site_url')
+SITE_TITLE = config.get(u'ckan.site_title', u'CKAN')
+
+
+def _package_search(data_dict):
+    """
+    Helper method that wraps the package_search action.
+
+     * unless overridden, sorts results by metadata_modified date
+     * unless overridden, sets a default item limit
+    """
+    context = {u'model': model, u'session': model.Session,
+               u'user': g.user, u'auth_user_obj': g.userobj}
+    if u'sort' not in data_dict or not data_dict['sort']:
+        data_dict['sort'] = u'metadata_modified desc'
+
+    if u'rows' not in data_dict or not data_dict['rows']:
+        data_dict['rows'] = ITEMS_LIMIT
+
+    # package_search action modifies the data_dict, so keep our copy intact.
+    query = logic.get_action(u'package_search')(context, data_dict.copy())
+
+    return query['count'], query['results']
+
+
+def output_feed(results, feed_title, feed_description, feed_link, feed_url,
+                navigation_urls, feed_guid):
+    author_name = config.get(u'ckan.feeds.author_name', u'').strip() or \
+        config.get(u'ckan.site_id', u'').strip()
+    author_link = config.get(u'ckan.feeds.author_link', u'').strip() or \
+        config.get(u'ckan.site_url', u'').strip()
+
+    # TODO: language
+    feed_class = None
+    for plugin in plugins.PluginImplementations(plugins.IFeed):
+        if hasattr(plugin, u'get_feed_class'):
+            feed_class = plugin.get_feed_class()
+
+    if not feed_class:
+        feed_class = _FixedAtomFeed
+
+    feed = feed_class(title=feed_title,
+                      url=feed_link,
+                      language=u'en',
+                      author={u'name': author_name},
+                      author_link=author_link,
```

Comment:
There is no `author_link`, use `uri` in the `author` dict

Reply:
done